subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_shi_ite.c

```diff
@@ -63,23 +63,67 @@ struct shi_it8xxx2_cfg {
 	const struct gpio_dt_spec cs;
 };
 
+enum shi_ite_pm_policy_state_flag {
+	SHI_ITE_PM_POLICY_FLAG,
+	SHI_ITE_PM_POLICY_FLAG_COUNT,
+};
+
 struct shi_it8xxx2_data {
 	/* Peripheral data */
 	struct ec_host_cmd_rx_ctx *rx_ctx;
 	struct ec_host_cmd_tx_buf *tx;
 	struct gpio_callback cs_cb;
+	struct k_work_delayable cs_off_put;
 	/* Current state */
 	enum shi_state_machine shi_state;
 	/* Buffers */
 	uint8_t in_msg[SPI_RX_MAX_FIFO_SIZE] __aligned(4);
 	uint8_t out_msg[SPI_TX_MAX_FIFO_SIZE] __aligned(4);
+	ATOMIC_DEFINE(pm_policy_state_flag, SHI_ITE_PM_POLICY_FLAG_COUNT);
 };
 
 struct ec_host_cmd_shi_ite_ctx {
 	/* SHI device instance */
 	const struct device *dev;
 };
 
+static void shi_ite_pm_policy_state_lock_get(struct shi_it8xxx2_data *data,
+					     enum shi_ite_pm_policy_state_flag flag)
+{
+	if (atomic_test_and_set_bit(data->pm_policy_state_flag, flag) == 0) {
+		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+		k_work_reschedule(&data->cs_off_put, K_SECONDS(3));
+	}
+}
+
+static void shi_ite_pm_policy_state_lock_put(struct shi_it8xxx2_data *data,
+					     enum shi_ite_pm_policy_state_flag flag)
+{
+	if (atomic_test_and_clear_bit(data->pm_policy_state_flag, flag) == 1) {
+		pm_policy_state_lock_put(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+		k_work_cancel_delayable(&data->cs_off_put);
+	}
+}
+
+/*
+ * When AP is In S0 state, AP assert CS of SPI for a new command transaction.
+ * Under the condition, SoC will not enter deep sleep until AP de-assert the CS.
+ * But If AP is powered off (G3 state) CS will go low (assert CS).
```

Comment:
Am I understanding correctly, that CS always goes low (asserted) if AP enters G3?

Reply:
Right.

Reply:
So there is only internal PU in AP, which is disconnected when AP enters G3? Despite this change, I'm wondering how it is possible that CS goes low, shouldn't PU be somewhere?

Reply:
If I understand correctly, no external PU on AP_SPI_EC_CS_L.
And AP's power rail will be cut off in G3 state, so internal PU will not work.

Reply:
Is there any possibility to have internal PU on the EC side? I've been working with stm32 (SPI) and they do it [this way](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/dts/st/h7/stm32h743vitx-pinctrl.dtsi#L2079). Additionally, I can see that ITE also has the pull-up for some pins with an alternate function, e.g. [keyboard](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/riscv/ite/it8xxx2-pinctrl-map.dtsi#L115).

Reply:
On IT8xxx2, enable internal PU on EC's CS will pull this pin up to PP1800_EC_VCC connected to PP1800_VIO18_S3. PP1800_VIO18_S3 is turned off in G3 state. (Please see the attachment)

![ec-vcc-1p8v](https://github.com/user-attachments/assets/5ba58f5f-2757-4ac2-845c-8c897242f212)

Reply:
Thanks for the explanation. IMO, there always should be PU for CS pin, regardless of AP power state. But it is hardware change. Software change is good anyway.